### PR TITLE
Add Count rule

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #185

### CODE REVIEW

**Comment by arogachev** [comment]
If `$exactly` concept is OK, it can be later added to `HasLength` rule too.

File: src/Rule/Count.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+use Attribute;
+use Countable;
+use InvalidArgumentException;
+use Yiisoft\Validator\FormatterInterface;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\Rule;
+use Yiisoft\Validator\ValidationContext;
+use function count;
+
+#[Attribute(Attribute::TARGET_PROPERTY)]
```
Comment (rustamwin) [inline src/Rule/Count.php:16]: I think it's better to place it after docs-block

File: src/Rule/Count.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+namespace Yiisoft\Validator\Rule;
+
+use Attribute;
+use Countable;
+use InvalidArgumentException;
+use Yiisoft\Validator\FormatterInterface;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\Rule;
+use Yiisoft\Validator\ValidationContext;
+use function count;
```
Comment (rustamwin) [inline src/Rule/Count.php:14]:
```suggestion

use function count;
```

File: src/Rule/Count.php [hunk 3]
```diff
@@ -49,11 +49,18 @@ public function __construct(
         /**
          * @var string user-defined error message used when the number of items is smaller than {@see $min}.
          */
-        private string $tooFewItemsMessage = 'This value should contain at least {min, number} {min, plural, one{items} other{items}}.',
+        private string $tooFewItemsMessage = 'This value must contain at least {min, number} ' .
+        '{min, plural, one{items} other{items}}.',
```
Comment (samdark) [inline src/Rule/Count.php:52]:
```suggestion
        '{min, plural, one{item} other{items}}.',
```

**Review by samdark — APPROVED** [review]

**Review by rustamwin — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**